### PR TITLE
upload files into dropbox

### PR DESCRIPTION
**Pull request by JEdilbertoTC**
upload files into dropbox

### CODE REVIEW

File: backend/src/main/java/org/uresti/pozarreal/service/impl/FilesStorageServiceImpl.java [hunk 1]
```diff
@@ -63,6 +82,41 @@ public Path save(MultipartFile file) {
...
+
+            paymentEvidence.setId(UUID.randomUUID().toString());
+            paymentEvidence.setPaymentId(paymentId);
+            paymentEvidence.setUrl(sharingRequests.getUrl());
+
+            paymentEvidenceRepository.save(paymentEvidence);
+
+            return sharingRequests.getUrl();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not store the file. Error: " + e.getMessage());
```
Comment (iuresti) [inline backend/src/main/java/org/uresti/pozarreal/service/impl/FilesStorageServiceImpl.java:116]: throw instead a PozarrealSystemException.. you can see an example in DBUpdaterServiceImpl... When you catch a exception and throw other instead remember always to nest the exceptions, so the stack trace can contain the root cause always

File: frontend/src/app/components/circuitos/circuitos.component.ts [hunk 2]
```diff
@@ -110,6 +112,12 @@ export class CircuitosComponent implements OnInit {
         bimesterPayment.validated = payment.validated;
         bimesterPayment.amount += this.newPayment.amount;
         bimesterPayment.complete = this.maintenanceFee <= bimesterPayment.amount;
+        const files = this.newPayment.files;
+        if (files) {
+          for (let i = 0; i < files.length; i++) {
```
Comment (iuresti) [inline frontend/src/app/components/circuitos/circuitos.component.ts:117]: prefer always the "for each" syntax over the traditional for sintax

File: frontend/src/app/components/payments/payments.component.ts [hunk 3]
```diff
@@ -99,6 +101,12 @@ export class PaymentsComponent implements OnInit {
       console.log('Saving payment');
       console.log(this.newPayment);
       this.paymentService.save(this.newPayment).subscribe(() => {
+        const files = this.newPayment.files;
+        if (files) {
+          for (let i = 0; i < files.length; i++) {
```
Comment (iuresti) [inline frontend/src/app/components/payments/payments.component.ts:106]: Use for each syntax (not sure but probably it is a hint in the IDE)

**Review by iuresti — CHANGES_REQUESTED** [review]

File: frontend/src/app/components/circuitos/circuitos.component.ts (hunk 2 above)
Comment (JEdilbertoTC) [reply frontend/src/app/components/circuitos/circuitos.component.ts:117]: when I wrote this code i tried to use foreach method or map method but i think it’s no available or isn't iterable to files

**Review by iuresti — APPROVED** [review]

**Pull request merged**